sharding-proxy/sharding-proxy-backend/src/test/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/DatabaseCommunicationEngineFactoryTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.shardingproxy.backend.communication.jdbc;
+
+import org.apache.shardingsphere.shardingproxy.backend.communication.DatabaseCommunicationEngine;
+import org.apache.shardingsphere.shardingproxy.backend.communication.DatabaseCommunicationEngineFactory;
+import org.apache.shardingsphere.shardingproxy.backend.communication.jdbc.connection.BackendConnection;
+import org.apache.shardingsphere.shardingproxy.backend.schema.LogicSchema;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+public final class DatabaseCommunicationEngineFactoryTest {
+    
+    @Test
+    public void assertNewTextProtocolInstance() {
+        DatabaseCommunicationEngine engine = DatabaseCommunicationEngineFactory.getInstance().newTextProtocolInstance(mock(LogicSchema.class), "schemaName", mock(BackendConnection.class));
+        assertNotNull(engine);
+        assertThat(engine, instanceOf(JDBCDatabaseCommunicationEngine.class));
+    }
+    
+    @Test
+    public void assertNewBinaryProtocolInstance() {
+        DatabaseCommunicationEngine engine = DatabaseCommunicationEngineFactory.getInstance()
+                .newBinaryProtocolInstance(mock(LogicSchema.class), "schemaName", Collections.emptyList(), mock(BackendConnection.class));
+        assertNotNull(engine);
+        assertThat(engine, instanceOf(JDBCDatabaseCommunicationEngine.class));
+    }
+}
```

Comment:
Files should be end with a empty line.

Reply:
It seems that the last blank line is hidden in the changed files page. 

In IDE, the last blank lines is visible. 
![image](https://user-images.githubusercontent.com/6405801/66627002-dbb06f00-ec2c-11e9-8476-8a421636d45f.png)

And my merged pr's changed files page also does not show the last blank line, as seen in pr: https://github.com/apache/incubator-shardingsphere/pull/3204/files